### PR TITLE
test(download-file.js): Add some unit tests for download-file.js

### PR DESCRIPTION
**Pull request by alejandronanez**

### CODE REVIEW

File: test/download-file.spec.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import expect from 'expect';
+import sinon from 'sinon';
+import fs from 'fs';
+
+import { writeImageOnDisk } from '../src/download-file';
+
+/* eslint-disable  func-names, prefer-arrow-callback */
+describe('writeImageOnDisk', function() {
```
Comment (alejandronanez) [inline test/download-file.spec.js]: Not using arrow functions here.
We don't want to have this kind of problems
```shell
TypeError: Cannot set property 'sandbox' of undefined
```

Comment (andresdavid90) [inline test/download-file.spec.js]: Can you please explain a little bit more the problem? I believe you, but I want to understand it for myself. 🙃

File: package.json [hunk 2]
```diff
@@ -46,7 +46,8 @@
     "istanbul": "0.4.2",
     "mocha": "2.4.5",
     "nyc": "7.0.0",
-    "semantic-release": "^4.3.5"
+    "semantic-release": "^4.3.5",
+    "sinon": "1.17.4"
```
Comment (andresdavid90) [inline package.json]: I've never used `sinon` before. Can you please explain in what cases is it useful? Btw, I'm going out right now but I'll review this a little further in a few hours.

File: test/download-file.spec.js (hunk 1 above)
Comment (alejandronanez) [inline test/download-file.spec.js]: Sure. So basically we don't want to lose context when we do the tests. You can take a look at: [Arrow functions in mocha](http://mochajs.org/#arrow-functions)

File: test/download-file.spec.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  });
+
+  afterEach(function() {
+    this.sandbox.restore();
+  });
+
+  it('should set encoding', function() {
+    const { sandbox } = this;
+    const response = {
+      setEncoding: sandbox.stub(),
```
Comment (alejandronanez) [inline test/download-file.spec.js]: If we 'overwrite' the response `setEncoding` method with `sandbox.stub()` we will be able to check if it was called or not, also we can check how many times that method was called and know the arguments for each call.
That being said we will be 100% sure about what we pass and how we pass it.

File: test/download-file.spec.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  });
+
+  it('should call fs.mkdirSync with right params', function() {
+    const { sandbox } = this;
+    const response = {
+      setEncoding: sandbox.stub(),
+      on: sandbox.stub()
+    };
+
+    sandbox.stub(fs, 'existsSync').returns(false);
```
Comment (alejandronanez) [inline test/download-file.spec.js]: Here we're _forcing_ `fs.existsSync` to return `false` so we can test this code:
```js
if (!fs.existsSync(path.dirname(fileName))) {
    fs.mkdirSync(path.dirname(fileName));
}
```

File: package.json (hunk 2 above)
Comment (alejandronanez) [inline package.json]: So Sinon helps us to 'watch' / 'fake' methods in our code. I explained what it does with these 2 examples:
[Comment #1](https://github.com/andresdavid90/pizza-guy/pull/13/files#r71974311)
[Comment #2](https://github.com/andresdavid90/pizza-guy/pull/13/files#r71974322)

**Comment by alejandronanez** [comment]
Also we're using `sinon.sandbox` instead of `sinon.spy` & `sinon.stub` I think it's cleaner and simple. You can see more about it here: http://sinonjs.org/docs/#sandbox

File: test/download-file.spec.js (hunk 4 above)
Comment (andresdavid90) [inline test/download-file.spec.js]: Nice.

File: test/download-file.spec.js (hunk 3 above)
Comment (andresdavid90) [inline test/download-file.spec.js]: ✌🏻

**Pull request merged**